pkgs/test/test/runner/configuration/randomize_order_test.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@TestOn("vm")
+
+import 'package:test_descriptor/test_descriptor.dart' as d;
+
+import 'package:test/test.dart';
+
+import '../../io.dart';
+
+void main() {
+  test("shuffles test order when passed a seed", () async {
+    await d.file("test.dart", """
+      import 'package:test/test.dart';
+
+      void main() {
+        test("test 1", () {});
+        test("test 2", () {});
+        test("test 3", () {});
+        test("test 4", () {});
+      }
+    """).create();
+
+    // Test with a given seed
+    var test =
+        await runTest(["test.dart", "--test-randomize-ordering-seed=987654"]);
+    expect(
+        test.stdout,
+        containsInOrder([
+          "+0: test 4",
+          "+1: test 3",
+          "+2: test 1",
+          "+3: test 2",
+          "+4: All tests passed!"
+        ]));
+    await test.shouldExit(0);
+
+    // Do not shuffle when passed 0
+    test = await runTest(["test.dart", "--test-randomize-ordering-seed=0"]);
+    expect(
+        test.stdout,
+        containsInOrder([
+          "+0: test 1",
+          "+1: test 2",
+          "+2: test 3",
+          "+3: test 4",
+          "+4: All tests passed!"
+        ]));
+    await test.shouldExit(0);
+
+    // Do not shuffle when passed nothing
+    test = await runTest(["test.dart"]);
+    expect(
+        test.stdout,
+        containsInOrder([
+          "+0: test 1",
+          "+1: test 2",
+          "+2: test 3",
+          "+3: test 4",
+          "+4: All tests passed!"
+        ]));
+    await test.shouldExit(0);
+
+    // Shuffle when passed random
+    // TODO(kathrins): How can to mock the behaviour of random?
```

Comment:
It's going to be a little difficult to mock the random because the internal logic isn't exposed cleanly. I'd just confirm that using `random` results in logging a non-zero seed 32 uint seed.